src/FocusApp.Client/Views/Social/UserBadgesPage.cs

```diff
@@ -0,0 +1,235 @@
+
+namespace FocusApp.Client.Views.Social;
+
+internal class UserBadgesPage : BasePage
+{
+    private readonly IAPIClient _client;
+    private readonly IAuthenticationService _authenticationService;
+    private readonly PopupService _popupService;
+    private readonly FocusAppContext _localContext;
+
+    public Image _selectedCheckmark;
+
+    public Guid SelectedBadgeId { get; set; }
+
+    public FlexLayout _flexLayout;
+    public Dictionary<Badge, Image> BadgeDict { get; set; }
+
+    public UserBadgesPage(IAPIClient client, IAuthenticationService authenticationService, PopupService popupService, FocusAppContext localContext)
+    {
+        _client = client;
+        _authenticationService = authenticationService;
+        _popupService = popupService;
+        _localContext = localContext;
+
+        // FlexLayout for the Badges
+        _flexLayout = new FlexLayout
+        {
+            Direction = FlexDirection.Row,
+            Wrap = FlexWrap.Wrap,
+            JustifyContent = FlexJustify.SpaceAround
+        };
+
+        Content = new Grid
+        {
+            RowDefinitions = Rows.Define(80, 10, Star, 80),
+            ColumnDefinitions = Columns.Define(Star, Star),
+            BackgroundColor = AppStyles.Palette.DarkMauve,
+
+            Children =
+            {
+            // Header
+            new Label
+            {
+                Text = "Badges",
+                TextColor = Colors.Black,
+                FontSize = 40
+            }
+            .Row(0)
+            .Column(0)
+            .Center()
+            .ColumnSpan(2)
+            .CenterVertical()
+            .Paddings(top: 5, bottom: 5, left: 5, right: 5),
+
+            // Back Button
+            new Button
+            {
+                Text = SolidIcons.ChevronLeft,
+                TextColor = Colors.Black,
+                FontFamily = nameof(SolidIcons),
+                FontSize = 40,
+                BackgroundColor = Colors.Transparent
+            }
+            .Left()
+            .CenterVertical()
+            .Paddings(top: 10, bottom: 10, left: 15, right: 15)
+            .Column(0)
+            .Invoke(button => button.Released += (sender, eventArgs) =>
+            BackButtonClicked(sender, eventArgs)),
+
+            // Header & Content Divider
+            new BoxView
+            {
+                Color = Color.Parse("Black"),
+                WidthRequest = 400,
+                HeightRequest = 2,
+            }
+            .Bottom()
+            .Row(0)
+            .Column(0)
+            .ColumnSpan(5),
+
+            new ScrollView
+            {
+               Content = _flexLayout
+            }
+            // Badges FlexLayout
+            .ColumnSpan(2)
+            .Row(2)
+            }
+        };
+
+    }
+
+    protected override async void OnAppearing()
+    {
+        base.OnAppearing();
+
+        // Check if the user is logged in
+        if (_authenticationService.CurrentUser == null)
```

Comment:
For consistency, could we check if the user is logged in  the same way that's done in the SocialPage and ShopPage?
https://github.com/ChrispyPeaches/FocusFriends/blob/user-badges/src/FocusApp.Client/Views/Social/SocialPage.cs#L207-L211